web/src/connectors/task-annotator-connector/use-split-validation.ts

```diff
@@ -78,24 +81,54 @@ export default function useSplitValidation({
         { enabled: isSplitValidation }
     );
 
-    const userPages: AnnotationsByUserObj[] = useMemo(() => {
-        if (!byUser) {
+    const { data: pages } = useGetPageSummary(
+        { taskId: task?.id, taskType: task?.is_validation },
+        { enabled: Boolean(task) }
+    );
+
+    const { data: latestRevision } = useLatestAnnotationsByUser(
+        {
+            fileId,
+            jobId: job?.id,
+            pageNumbers: pages?.annotated_pages // передать сюда annotated из page sammury
+        },
+        { enabled: isSplitValidation }
+    );
+
+    const latestRevisionByAnnotators: UserRevision[] = useMemo(() => {
+        if (!latestRevision) {
             return [];
         }
-        return byUser[currentPage].filter((userPage) => userPage.user_id !== userId);
-    }, [byUser, currentPage]);
 
-    const taxonLabels = useAnnotationsTaxons(userPages);
+        const result = convertToRevisionByUserArray(latestRevision as unknown as LatestRevisionObj);
+        const currentUser = userId;
+
+        return result.filter((revision) => revision.user_id !== currentUser);
+    }, [latestRevision]);
+
+    const latestRevisionByCurrentUser: UserRevision[] = useMemo(() => {
+        if (!latestRevision) {
+            return [];
+        }
+
+        const result = convertToRevisionByUserArray(latestRevision as unknown as LatestRevisionObj);
+        const currentUser = userId;
+
+        return result.filter((revision) => revision.user_id === currentUser);
+    }, [latestRevision]);
+
+    const taxonLabels = useAnnotationsTaxons(latestRevisionByAnnotators);
 
     const { mapAnnotationPagesFromApi } = useAnnotationsMapper(taxonLabels, [byUser, taxonLabels]);
 
-    const annotationsByUserId = useMemo(() => {
+    const latestRevisionByAnnotatorsWithBounds = useMemo(() => {
+        if (!latestRevisionByAnnotators) return {};
         return mapAnnotationPagesFromApi(
             (page: AnnotationsByUserObj) => page.user_id,
-            userPages,
+            latestRevisionByAnnotators as unknown as AnnotationsByUserObj[],
```

Comment:
The same with strange data casting

Reply:
fixed https://github.com/epam/badgerdoc/pull/584/commits/79c03c121c6c6cbd531b96f01677773062a1bb65